packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -0,0 +1,316 @@
+        onError: plugin._onTimerError,
+      );
+    }
+    return plugin;
+  }
+
+  /// An [AWSLoggerPlugin] to use only for testing.
+  @protected
+  @visibleForTesting
+  CloudWatchLoggerPlugin.testPlugin({
+    required CloudWatchLogsClient client,
+    required CloudWatchLoggerPluginConfiguration pluginConfig,
+    required CloudWatchLogStreamProvider logStreamProvider,
+    required QueuedItemStore logStore,
+    RemoteLoggingConstraintProvider? remoteLoggingConstraintProvider,
+  })  : _syncing = false,
+        _enabled = pluginConfig.enable,
+        _pluginConfig = pluginConfig,
+        _logStore = logStore,
+        _remoteLoggingConstraintProvider = remoteLoggingConstraintProvider,
+        _logStreamProvider = logStreamProvider,
+        _clinet = client;
+
+  CloudWatchLoggerPlugin._({
+    required AWSCredentialsProvider credentialsProvider,
+    required CloudWatchLoggerPluginConfiguration pluginConfig,
+    RemoteLoggingConstraintProvider? remoteLoggingConstraintProvider,
+    CloudWatchLogStreamProvider? logStreamProvider,
+    QueuedItemStore? logStore,
+  })  : _syncing = false,
+        _enabled = pluginConfig.enable,
+        _pluginConfig = pluginConfig,
+        _logStore = logStore ?? InMemoryQueuedItemStore(),
+        _remoteLoggingConstraintProvider = remoteLoggingConstraintProvider ??
+            (pluginConfig.defaultRemoteConfiguration != null
+                ? DefaultRemoteLoggingConstraintProvider(
+                    config: pluginConfig.defaultRemoteConfiguration!,
+                    credentialsProvider: credentialsProvider,
+                  )
+                : null),
+        _clinet = CloudWatchLogsClient(
+          region: pluginConfig.region,
+          credentialsProvider: credentialsProvider,
+        ),
+        _logStreamProvider = logStreamProvider ??
+            DefaultCloudWatchLogStreamProvider(
+              logGroupName: pluginConfig.logGroupName,
+              client: CloudWatchLogsClient(
+                region: pluginConfig.region,
+                credentialsProvider: credentialsProvider,
+              ),
+            );
+
+  final CloudWatchLoggerPluginConfiguration _pluginConfig;
+  final CloudWatchLogsClient _clinet;
+  final CloudWatchLogStreamProvider _logStreamProvider;
+  final QueuedItemStore _logStore;
+  bool _syncing;
+  bool _enabled;
+  StoppableTimer? _timer;
+  RemoteLoggingConstraintProvider? _remoteLoggingConstraintProvider;
+
+  set remoteLoggingConstraintProvider(
+    RemoteLoggingConstraintProvider remoteProvider,
+  ) {
+    if (_remoteLoggingConstraintProvider != null) {
+      logger.error(
+        'remoteLoggingConstraintProvider is already configured.',
+      );
+      return;
+    }
+    _remoteLoggingConstraintProvider = remoteProvider;
+  }
+
+  Future<void> _startSyncingIfNotInProgress() async {
+    Future<void> startSyncing() async {
+      String logStream;
+      try {
+        logStream = await _logStreamProvider.logStream;
```

Comment:
Hmm okay. Calling this every 60 seconds seems kinda wasteful. Could you not react to its deletion when uploading log events?

Reply:
You probably have to do that anyway so I would make that the default mechanism instead of trying to recreate the log stream for each batch.

If a call to `PutLogEvents` fails with a `ResourceNotFoundException` you should be attempting to recreate the log stream anyway no?